Update GRO520 for A25

MarcOlivierFecteau opened the pull request:
This commit (completely) updates the baseline package for GRO520.

NOTES:

- This commit does NOT contain any solution: only the Python and ROS2 boilerplate code is present, along with some notes and instructions.

CHANGES:

- Remove duplicate function `yaw_from_quaternion`;
- Modify comments for more explicit instructions;
- Use Python's good practices for `main` boilerplate code;
- Remove unnecessary imports;
- Use more explicit names where applicable;
- Use `try-except-finally` block for a more graceful shutdown of ROS2 nodes;
- Add a versioning schema for ulterior versions of the packages;
- Replace comments for docstrings where applicable;

Commit:
Update GRO520 for A25

This commit (completely) updates the baseline package for GRO520.

NOTES:

- This commit does NOT contain any solution: only the Python and ROS2
  boilerplate code is present, along with some notes and instructions.

CHANGES:

- Remove duplicate function `yaw_from_quaternion`;
- Modify comments for more explicit instructions;
- Use Python's good practices for `main` boilerplate code;
- Remove unnecessary imports;
- Use more explicit names where applicable;
- Use `try-except-finally` block for a more graceful shutdown of ROS2
  nodes;
- Add a versioning schema for ulterior versions of the packages;
- Replace comments for docstrings where applicable;

chameau5050 commented on racecar_beacon/package.xml:
```diff
@@ -2,23 +2,25 @@
-  <version>0.0.0</version>
-  <description>TODO: Package description</description>
-  <maintainer email="lali3401@usherbrooke.ca">racecar</maintainer>
+  <version>2025.4.12</version>
```
This does not realy conform to the classic versioning use in software devellopement (major.minor.fix) we can start at 1.0.0 for now

chameau5050 commented on racecar_beacon/package.xml:
```diff
@@ -2,23 +2,25 @@
 
+  <buildtool_depend>ament_python</buildtool_depend>
+
+  <exec_depend>rclpy</exec_depend>
+  <exec_depend>geometry_msgs</exec_depend>
+  <exec_depend>nav_msgs</exec_depend>
+  <exec_depend>sensor_msgs</exec_depend>
+  <exec_depend>tf_transformations</exec_depend>
```
I know this is not your change but since we thouching this can you sort the dependecy in alphabetic other

chameau5050 commented on racecar_beacon/racecar_beacon/ros_monitor.py:
```diff
@@ -1,66 +1,71 @@
-import socket
-import threading
-from struct import *
+from .lab_poll_pos import yaw_from_quaternion
```
A would prefer to define a utils file at the place of using code from a lab
```suggestion
from racecar_beacon.utils import yaw_from_quaternion
```

chameau5050 commented on racecar_beacon/setup.py:
```diff
@@ -1,27 +1,34 @@
     name=package_name,
-    version='0.0.0',
-    packages=find_packages(exclude=['test']),
+    version="2025.4.12",
```
same comment for version num

chameau5050 reviewed (changes requested):
Good job ! Only a couple change needed

Commit:
Apply suggested modifications

CHANGES:

- Change from `yyyy.mm.dd` version schema to SemVer;
- Move common functions into `utils.py`;
- Sort dependencies alphabetically in `package.xml`.
